### PR TITLE
DX: check deprecations exactly

### PR DESCRIPTION
**Pull request by keradus**
recovers #7610

### CODE REVIEW

File: tests/TestCase.php [hunk 1]
```diff
@@ -35,12 +35,16 @@ abstract class TestCase extends BaseTestCase
...
     {
         if (null !== $this->previouslyDefinedErrorHandler) {
-            foreach ($this->expectedDeprecations as $expectedDeprecation) {
-                self::assertContains($expectedDeprecation, $this->actualDeprecations);
-            }
+            $this->actualDeprecations = array_unique($this->actualDeprecations);
+            sort($this->actualDeprecations);
+            $this->expectedDeprecations = array_unique($this->expectedDeprecations);
+            sort($this->expectedDeprecations);
+            self::assertSame($this->expectedDeprecations, $this->actualDeprecations);
```
Comment (Wirone) [inline tests/TestCase.php:42]: [My question](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7610#discussion_r1434440807) still stands. After this change tests' result is conditioned by implementation. Let's say `Foo::bar()` triggers deprecation, if our logic uses this, then we expect the deprecation in tests. Why would we need to modify expectation in test when that method is used again in the tested logic? Does the fact that deprecated method is used multiple times change anything?

Comment (keradus) [reply tests/TestCase.php:42]: we care which deprecations were triggered, not how many times nor in which order
(thus, we have `array_uniq` and `sort`)

Comment (Wirone) [reply tests/TestCase.php:42]: Hmmmm, probably I overlooked `array_unique()` 🤔. So how is this different from what we have currently? I believe with the previous loop it also was checking if all expected deprecations were caught, so do we want to check if there were more expectations than actual deprecations, or other way around?

Comment (keradus) [reply tests/TestCase.php:42]: current master is checking if all deprecations listed in `expectDeprecation` happen, but not checking if any more happen on top. This PR makes sure we check both.

When using Sf brigde, we checked for both.

Comment (keradus) [reply tests/TestCase.php:42]: ie master is not failing without https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7742/files#diff-80e18dd53b2dd28618278f183c30a01bc63e8aceec7508ab815ed2f124aa8f3dR58 , but this PR is.

**Review by Wirone — APPROVED** [review]

**Comment by keradus** [comment]
thank you @kubawerlos

**Pull request merged**